ast/ast.fbs

```diff
@@ -24,6 +25,11 @@ table ArrayType {
   element:MonoType;
 }
 
+table VectorType {
+  base_node:BaseNode;
+  element:MonoType;
+}
+
```

Comment:
Actually I don't think support for `VectorType` is needed in AST at all.

In AST, we use types only as part of Flux syntax for `builtin` statements, which declare the type of a thing that is actually implemented in Go (not Flux).

For example, on this line everything after `:` is the type:
https://github.com/influxdata/flux/blob/master/stdlib/universe/universe.flux#L15

However, for `vector` so far we don't expect to need to declare any builtins that use a `vector` type.  So you don't need to do anything with AST for this PR. It can be completely about create a new Vector monotype for semantic graph only.